src/databases/CGNS/avtCGNSFileReader.C

```diff
@@ -1777,350 +1782,1018 @@ avtCGNSFileReader::GetUnstructuredMesh(int timestate, int base, int zone, const
-                        verts[7] = elem[7]-1;
-                        ugrid->InsertNextCell(VTK_HEXAHEDRON, 8, verts);
-                        higherOrderWarning = true;
-                        elem += 20;
-                        break;
-                    case HEXA_27:
-                        verts[0] = elem[0]-1;
-                        verts[1] = elem[1]-1;
-                        verts[2] = elem[2]-1;
-                        verts[3] = elem[3]-1;
-                        verts[4] = elem[4]-1;
-                        verts[5] = elem[5]-1;
-                        verts[6] = elem[6]-1;
-                        verts[7] = elem[7]-1;
-                        ugrid->InsertNextCell(VTK_HEXAHEDRON, 8, verts);
-                        higherOrderWarning = true;
-                        elem += 27;
-                        break;
-                    default:
-                        delete [] coords[0];
-                        delete [] coords[1];
-                        delete [] coords[2];
-                        delete [] elements;
-                        elements = 0;
-                        ugrid->Delete();
-                        EXCEPTION1(InvalidVariableException, meshname);
-                        break;
-                    }
-                }
+        vtkIdType verts[27];
+        const cgsize_t *zonePointsPtr = zonePoints;
+
+        //
+        // Iterate over our polygons, and add them to the mesh.
+        //
+        for (cgsize_t c = 0; c < numSectionZones; ++c)
+        {
+            //
+            // The connectivity offsets have size numSectionZones + 1
+            // so that we can grab the number of points for each zone.
+            //
+            int numVertices = connOffsets[c+1] - connOffsets[c];
 
-                debug4 << mName << "Done reading cell connectivity." << endl;
-                delete [] elements;
+            for (int v = 0; v < numVertices; ++v)
+            {
+                verts[v] = zonePointsPtr[v]-1;
             }
 
-            // Tell the user if we found any higher order elements.
-            if(higherOrderWarning)
+            ugrid->InsertNextCell(VTK_POLYGON, numVertices, verts);
+            zonePointsPtr += numVertices;
+        }
+
+        delete [] zonePoints;
+        delete [] connOffsets;
+    }
+}
+
+// ****************************************************************************
+//  Method: avtCGNSFileReader::ReadNGonAndNFaceSections
+//
+//  Purpose:
+//      NFace_n sections are used to define arbitrary polyhedra, and they are
+//      always used in conjunction with NGon_n. NFace_n "elements" define the
+//      zone-to-face connectivity, and NGon_n "elements" define the
+//      face-to-point connectivity.
+//
+//  Arguments:
+//      urgrid                  The unstructured grid to add our cells to.
+//      meshName                The name of our mesh.
+//      mixedAndNamedSections   An array of our mixed and named section indices.
```

Comment:
This should describe nGonSections, nFaceSections.

Reply:
Done.

Reply:
Remove mixedAndNamedSections.